shared/gateway/api_middleware.go

```diff
@@ -356,10 +360,102 @@ func (m *ApiProxyMiddleware) handleGetBeaconStateSsz(endpoint string, writer htt
 	}
 }
 
+func (m *ApiProxyMiddleware) handleGetBeaconBlockSsz(endpoint string, writer http.ResponseWriter, request *http.Request) {
+	// Prepare request values for proxying.
+	request.URL.Scheme = "http"
+	request.URL.Host = m.GatewayAddress
+	request.RequestURI = ""
+	request.URL.Path = "/eth/v1/beacon/blocks/{block_id}/ssz"
+	handleUrlParameters(endpoint, request, writer, []string{})
+
+	// Proxy the request to grpc-gateway.
```

Comment:
Can you abstract this to a common method? It seams the beacon state method shares a lot of the same lines of code.

Reply:
This had been written before #8984 and duplicated on purpose because I was not sure what shape will the refactor have. The code for handling beacon block SSZ lives in `endpoints.go` now and is no longer duplicated.